### PR TITLE
Enhancement/chat service

### PR DESCRIPTION
**Pull request by Venkataraochavala**
Enhanced chat_service.py for better field extraction, validation, and mid-flow corrections.
Refined normalization rules and response logic in chat_service_prompt.txt.
Improved test coverage with new and edge case scenarios in test_chat_assistant.py.
Updated chat_api.py to align with revised chat service interfaces.
Refactored ads_api.py for improved payload construction and error handling.

### CODE REVIEW

File: services/chat_service.py [hunk 1]
```diff
@@ -68,67 +21,112 @@ def start_session():
...
+        session["campaign_data"]["loginCustomerId"] = login_customer_id.replace("-", "")
+        message = f"{message} (My customer id is {login_customer_id})"
+
     # Expiry check
     if datetime.now(timezone.utc) - session["last_activity"] > SESSION_TIMEOUT:
         del sessions[session_id]
-        return JSONResponse(content={"status": "error", "message": "Session expired."}, status_code=400)
+        return JSONResponse(
+            content={"status": "error", "message": "Session expired."},
+            status_code=400
```
Comment (Kailash-Kumar04) [inline services/chat_service.py:85]: change status to 401

File: services/chat_service.py [hunk 2]
```diff
@@ -68,67 +21,112 @@ def start_session():
...
+    for new_data in extracted_jsons:
+        validated_data.update(validate_and_clean_data(new_data))
+
+    # Update session with only validated data
+    collected_data.update(validated_data)
+
+    # Special handling for duration to calculate end date
+    if "durationDays" in validated_data:
         try:
-            new_data = json.loads(json_match.group())
```
Comment (Kailash-Kumar04) [inline services/chat_service.py:97]: using a pydantic model will save us from this

File: tests/test_chat_assistant.py [hunk 3]
```diff
@@ -0,0 +1,1331 @@
...
+import json
+import time
+from datetime import datetime
+from typing import Dict, Any
+from colorama import init, Fore, Style
+
+# Initialize colorama for colored output
+init(autoreset=True)
+
+BASE_URL = "http://127.0.0.1:8000"
```
Comment (Kailash-Kumar04) [inline tests/test_chat_assistant.py:16]: add it into .env

File: apis/ads_api.py [hunk 4]
```diff
@@ -195,12 +196,23 @@ class GoogleNegativeRequest(BaseModel):
 async def gks_positive(
         google_keyword_request: GoogleKeywordsRequest,
         client_code: str = Header(..., alias="clientCode"),
+        session_id: str = Header(..., alias="sessionId")
 ):
     try:
+        if session_id not in sessions:
```
[the remark this answers is not in the document]
Comment (Venkataraochavala) [reply apis/ads_api.py:202]: moved this check into the service

File: tests/test_chat_assistant.py (hunk 3 above)
Comment (Venkataraochavala) [reply tests/test_chat_assistant.py:16]: i have this in the env file as base url
